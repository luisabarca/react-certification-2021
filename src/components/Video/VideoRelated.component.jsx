import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const VideoContainer = styled.div`
  display: flex;
  background-color: white;
  flex-direction: row;
`;

const VideoTitle = styled.h2`
  font-size: 0.9rem;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  color: black;
  font-weight: 300;
  letter-spacing: 0.001em;
  padding: 5px 5px 0 5px;
  margin: 0;
  flex: 2;
`;

const VideoThumbnail = styled.div`
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  background-image: url(${(props) => props.url || ''});
  height: 90px;
  width: 120px;
  flex: 1;
`;

const VideoRelated = ({ video }) => {
  const { thumbnails, title } = video.snippet;
  const id = video.id.videoId || video.id.channelId;

  return (
    <Link to={`/video?id=${id}`}>
      <VideoContainer>
        <VideoThumbnail role="img" url={thumbnails.default.url} />
        <VideoTitle>{title}</VideoTitle>
      </VideoContainer>
    </Link>
  );
};

export default VideoRelated;
