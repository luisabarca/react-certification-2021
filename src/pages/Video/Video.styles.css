
.videopage {
    flex: 1;
    display: flex;
    margin: 1rem auto;
    justify-content: center;
}

.video-player {
    flex: 1;
}

.videopage > div {
    padding: 1rem;
}

.videopage > div:first-child {
    flex: 3;
}

.videopage > div:last-child {
    flex: 1;
}

.videopage iframe {
    border: none;
    flex: 1;
    width: 100%;
    min-height: 480px;
    height: auto;
}

.statistics {
    flex: 1;
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-size: 0.9rem;
}

.statistics > span {
    flex: 1;
}

.statistics > span:first-child{
    flex: 8;
}