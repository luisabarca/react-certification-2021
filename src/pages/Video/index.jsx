/* eslint-disable react/no-danger */
import React from 'react';
import { useLocation } from 'react-router-dom';
import { ThumbDown, ThumbUp } from '@material-ui/icons';
import VideoRelated from '../../components/Video/VideoRelated.component';
import { useVideo, useRelatedVideos } from '../../utils/hooks/useVideos';
import './Video.styles.css';

function VideoPage() {
  const query = new URLSearchParams(useLocation().search);
  const videoId = (query && query.get('id')) || '';

  // Get video from YT.
  const video = useVideo(videoId);
  const related = useRelatedVideos(videoId);

  if (!video) return null;

  const {
    snippet: { title, description },
    statistics,
    player,
  } = video;

  return (
    <section className="videopage">
      <div className="video-main">
        <div
          className="video-player"
          dangerouslySetInnerHTML={{
            __html: player.embedHtml,
          }}
        />

        <h3>{title}</h3>

        <div className="statistics">
          <span>{statistics.viewCount} views</span>
          <span>
            <ThumbUp /> {statistics.likeCount}
          </span>
          <span>
            <ThumbDown /> {statistics.dislikeCount}
          </span>
        </div>

        <p>{description}</p>
      </div>
      <div className="video-sidebar">
        {related &&
          related.map((item) => {
            return <VideoRelated video={item} />;
          })}
      </div>
    </section>
  );
}

export default VideoPage;
