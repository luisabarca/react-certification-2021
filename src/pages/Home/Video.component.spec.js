/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Video from './Video.component';
import videos from '../../data/youtube-data.json';

describe('Video Component', () => {
  const video1 = videos.items[0];
  beforeEach(() => {
    render(<Video video={video1} />);
  });

  it('should match snapshot', () => {
    const snap = renderer.create(<Video video={video1} />);

    expect(snap.toJSON()).toMatchSnapshot();
  });

  it('should render a link', () => {
    const link = screen.getByRole('link');
    expect(link).toBeInTheDocument();
  });

  it('should render a thumbnail image', () => {
    const img = screen.getByRole('img');
    expect(img).toBeInTheDocument();
  });

  it('should render a title', () => {
    const title = screen.getByRole('heading');
    expect(title).toBeInTheDocument();
  });

  it('should render a description', () => {
    const text = screen.getByRole('note');
    expect(text).toBeInTheDocument();
  });
});
