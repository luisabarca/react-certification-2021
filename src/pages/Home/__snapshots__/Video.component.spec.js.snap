// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Component should match snapshot 1`] = `
<div
  role="link"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="sc-dlnjwi efxPah"
      role="img"
    />
    <h2
      className="sc-bdnxRM llyFNT"
    >
      Video Tour | Welcome to Wizeline Guadalajara
    </h2>
    <p
      className="sc-gtsrHT hDZSOc"
      role="note"
    >
      Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office.

In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, Mexico, home to over 400 employees. Wizeline has drawn attention for designing an office and culture focused on employee wellness and cultivating talent, earning a Super Espacios award by WeWork, Expansion Mexico, and Top Companies.

Read more about the office here: https://www.wizeline.com/creating-first-class-facilities/

If you're interested in working with Wizeline and want to experience the culture and tour our offices live, contact us at www.wizeline.com/contact/
    </p>
  </div>
</div>
`;
